datafusion/sql/src/planner.rs

```diff
@@ -228,12 +228,10 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         plan: LogicalPlan,
         alias: TableAlias,
     ) -> Result<LogicalPlan> {
-        let apply_name_plan = LogicalPlan::SubqueryAlias(SubqueryAlias::try_new(
-            plan,
-            self.normalizer.normalize(alias.name),
-        )?);
-
-        self.apply_expr_alias(apply_name_plan, alias.columns)
+        let plan = self.apply_expr_alias(plan, alias.columns)?;
```

Comment:
The order is changed so that we can get either `expr` or `table. expr`.

Example in array.slt
table.expr: `SELECT sum(data.a), sum(data.b) FROM unnest(make_array(1, 2, 3), make_array(4, 5, 6, 7)) as data(a, b);`
expr: `SELECT sum(a), sum(b) FROM unnest(make_array(1, 2, 3), make_array(4, 5, 6, 7)) as data(a, b);`